ignite/engine/__init__.py

```diff
@@ -41,6 +42,221 @@ def _prepare_batch(
     )
 
 
+def supervised_training_step(
+    model: torch.nn.Module,
+    optimizer: torch.optim.Optimizer,
+    loss_fn: Union[Callable, torch.nn.Module],
+    device: Optional[Union[str, torch.device]] = None,
+    non_blocking: bool = False,
+    prepare_batch: Callable = _prepare_batch,
+    output_transform: Callable = lambda x, y, y_pred, loss: loss.item(),
+) -> Callable:
+    """Factory function for supervised training.
+
+    Args:
+        model (torch.nn.Module): the model to train.
+        optimizer (torch.optim.Optimizer): the optimizer to use.
+        loss_fn (torch.nn loss function): the loss function to use.
+        device (str): device type specification (default: None).
```

Comment:
```suggestion
        device (str, optional): device type specification (default: None).
```
Please, have a look line 278-283 and update the docstring according to `create_supervised_trainer`. It misses `optional` to `device`, `non_blocking`, `prepare_batch` and `output_transform`.

This should be done for `supervised_training_step` and
* `supervised_training_step_amp` line 87
* `supervised_training_step_apex` line 147
* `supervised_training_step_tpu` line 195

Thank you !